worker/src/queues/evalQueue.ts

```diff
@@ -9,118 +9,64 @@ import {
-        QueueName.EvaluationExecution,
-        {
-          connection: connection,
-        }
-      )
-    : null;
+export class EvalExecutionQueue {
+  private static instance: Queue<
+    TQueueJobTypes[QueueName.EvaluationExecution]
+  > | null = null;
+
+  public static getInstance(): Queue<
+    TQueueJobTypes[QueueName.EvaluationExecution]
+  > | null {
+    if (EvalExecutionQueue.instance) return EvalExecutionQueue.instance;
+
+    const newRedis = createNewRedisInstance({ enableOfflineQueue: false });
+
+    EvalExecutionQueue.instance = newRedis
+      ? new Queue<TQueueJobTypes[QueueName.EvaluationExecution]>(
+          QueueName.EvaluationExecution,
+          {
+            connection: newRedis,
+            defaultJobOptions: {
+              removeOnComplete: true,
+              removeOnFail: 10_000,
+              attempts: 2,
+              backoff: {
+                type: "exponential",
+                delay: 5000,
+              },
+            },
+          },
+        )
+      : null;
 
-  return evalQueue;
-};
+    return EvalExecutionQueue.instance;
+  }
+}
 
 export const evalJobCreatorQueueProcessor = async (
-  job: Job<TQueueJobTypes[QueueName.TraceUpsert]>
+  job: Job<TQueueJobTypes[QueueName.TraceUpsert]>,
 ) => {
   try {
-    const startTime = Date.now();
-
-    const waitTime = Date.now() - job.timestamp;
-    recordIncrement("langfuse.queue.trace_upsert.request");
-    recordHistogram("langfuse.queue.trace_upsert.wait_time", waitTime, {
-      unit: "milliseconds",
-    });
-
     await createEvalJobs({ event: job.data.payload });
-
-    await getTraceUpsertQueue()
-      ?.count()
-      .then((count) => {
-        logger.debug(`Eval creation queue length: ${count}`);
-        recordGauge("trace_upsert_queue_length", count, {
-          unit: "records",
-        });
-        return count;
-      })
-      .catch();
-    recordHistogram(
-      "langfuse.queue.trace_upsert.processing_time",
-      Date.now() - startTime,
-      { unit: "milliseconds" }
-    );
     return true;
   } catch (e) {
     logger.error(
       `Failed job Evaluation for traceId ${job.data.payload.traceId}`,
-      e
+      e,
     );
     traceException(e);
     throw e;
   }
 };
 
 export const evalJobExecutorQueueProcessor = async (
-  job: Job<TQueueJobTypes[QueueName.EvaluationExecution]>
+  job: Job<TQueueJobTypes[QueueName.EvaluationExecution]>,
 ) => {
   try {
     logger.info("Executing Evaluation Execution Job", job.data);
-    const startTime = Date.now();
-
-    // reduce the delay from the time to get the actual wait time
-    // from the point where the job was ready to be processed
-    // reduce the delay by expo backoff: 2 ^ (attempts - 1) * delay, delay: 1000ms
-    const estimatedBackoffTime =
-      job.attemptsMade > 0 ? 1000 * Math.pow(2, job.attemptsMade - 1) : 0;
-
-    const normalisedWaitTime =
-      Date.now() -
-      job.timestamp -
-      (job.data.payload.delay ?? 0) -
-      estimatedBackoffTime;
```

Comment:
@maxdeichmann I'm not sure I understand why this was useful. Why would we take the retries into consideration for this type, but not the others? Do we need it at all?

Reply:
Confirmed offline that we'll try without.